### DISCUSSION TITLE
如何重置模型中心点坐标，到模型中央？

### DISCUSSION THREAD

**Discussion by geeklibin** [galacean/discussion#37] [Engine] [answered]
导入的gltf默认模型中心点都不固定。虽然我设置了他们的position为(0,0,0)，但是模型中心点位置不固定，所以放置位置都是满天飞。
![image](https://user-images.githubusercontent.com/9034445/221338533-5a629abc-767d-4a3a-a574-a00f48ba47f1.png)
例如上面的T恤模型的中心点位置是在模型的下方很远的位置。😭

**Comment by cptbtptpbcptdtptp** [comment]
这个是模型锚点设置的问题，有些模型局部（0，0，0）的位置不在中心点，当然你可以通过计算这个模型的包围盒，从而用包围盒的中心点让模型看起来居中，可以参考下方示例来计算模型的空间包围盒：
https://oasisengine.cn/#/examples/latest/input-glTFMerge

**Comment by cptbtptpbcptdtptp** [comment, marked as answer]
有三种办法：
- 让美术导出模型时就指定开发需要的模型中心点。
- 如果是特定模型，你可以在编辑器上明确这个模型的偏移坐标大小，通过放置在父节点上将这个偏移矫正回来。
- 如果是很多模型，你可以通过动态计算包围盒中心的方式，包围盒中心与原点的偏移，应该就是这个模型的偏移坐标大小，你可以通过放置在父节点上将这个偏移矫正回来。
> 下方为获取模型中心点的代码
``` typescript
const renderers = entity.getComponentsIncludeChildren(MeshRenderer, []);
    const boundingBox = renderers[0].bounds.clone();
    for (let i = renderers.length - 1; i > 0; i--) {
      BoundingBox.merge(boundingBox, renderers[i].bounds, boundingBox);
    }
    // 模型中心点
    boundingBox.getCenter(new Vector3());
```

建议让美术导出时指定模型中心点。
